### ISSUE TITLE
Add Delay Property to Bindings

### ISSUE THREAD

**Issue by joneuhauser** [AvaloniaUI/Avalonia#2589]
In WPF, it is possible to set a [delay](https://docs.microsoft.com/en-us/dotnet/api/system.windows.data.bindingbase.delay?view=netframework-4.8) to bindings. 

These delays are often extremely helpful to reduce the noise in the ViewModel: Say you want to bind the position of a slider to the viewmodel, which triggers a calculation intensive operation (takes at least a few hundred milliseconds). In this case, you don't want to trigger the calculation after each micro update of the slider, but only after major changes or the final value. 

Another use case is the following simple UI:
![grafik](https://user-images.githubusercontent.com/12698538/58552548-f56b8f80-8212-11e9-96cd-86cff9860cf3.png)
In this case, the user can edit either of the two `NumericUpDowns`. If the user enters a value in the top one, then the property `Square` is updated accordingly. If the user enters a value in the bottom one, it gets rounded to the nearest square number since the value in the top one has to be integer. This behavior is impossible if the "update" of the `Square` property is done without a delay: in this case, typing "8" would be rounded to 9, and typing "81" (as the user intended) is prevented. 

This feature is currently missing in Avalonia, so this issue proposes to add it.

**Comment by rossen-hristov** [comment]
We also need Binding.Delay a lot.

**Comment by maxkatz6** [comment]
Duplicate of https://github.com/AvaloniaUI/Avalonia/issues/7380
